hack/generator/spec/case-study-rolling-storage-versions.md

```diff
@@ -0,0 +1,936 @@
+# Case Study - Rolling Storage Versions
+
+This case study explores the recommended solution of using a *rolling storage version* where we update the storage schema of each resource each release of the service operator. We'll keep the storage version up to date with the latest GA release of each resource.
+
+For the purposes of discussion, we'll be following the version by version evolution of a theoretical ARM service that provides customer resource management (CRM) services. Synthetic examples are used to allow focus on specific scenarios one by one, providing motivation for specific features.
+
+Examples shown are deliberately simplified in order to focus, and therefore minutiae should be considered motivational, not binding. Reference the formal specification for precise details.
+
+# Version 2011-01-01 - Initial Release
+
+The initial release of the CRM includes a simple definition to capture information about a particular person:
+
+``` go
+package v20110101
+
+type Person struct {
+    Id        Guid
+    FirstName string
+    LastName  string
+}
+```
+
+We're not reusing the API version directly as our storage version. Instead, we define a separate (independent) type with a similar structure:
+
+``` go
+package v20110101storage
+
+type Person struct {
+    PropertyBag
+    FirstName   *string
+    Id          *Guid
+    LastName    *string
+}
+
+// Hub marks this type as a conversion hub.
+func (*Person) Hub() {}
+```
+
+Every property is marked as optional. Optionality doesn't matter at this point, as we currently have only single version of the API. However, as we'll see with later versions, forward and backward compatibility issues would arise if they were not optional.
```

Comment:
that sparked something in my mind, as we are moving to protobuf to define our API with this particular benefit in mind.
I did a quick googling, and I found this : 
https://github.com/istio/operator/tree/master/common-protos

Which seem to generate the proto files that map pretty much all CRDs of k8s and istio.
just an interesting parrallel with what we are trying to achieve here.